### PR TITLE
Tomcat9

### PR DESCRIPTION
**Pull request by kasecato**
I don't know the reason why socket check fails after start of Tomcat 9...
@bmuschko Could you fix it?

### CODE REVIEW

**Comment by bmuschko** [comment]
Thanks for providing the pull request. I am going to have a deeper look soon. Please remove the upgrade of the Gradle Wrapper from the change for now. I will do the upgrade independent of the Tomcat upgrade.

**Comment by k--kato** [comment]
@bmuschko Thanks for your support. I removed the upgrade of the Gradle Wrapper, Tomcat and version of project.

File: tomcat9x/src/main/groovy/com/bmuschko/gradle/tomcat/embedded/Tomcat9xServer.groovy [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+/**
+ * Tomcat 9x server implementation.
+ *
+ * @author Benjamin Muschko
+ * @author Andrey Bloschetsov
+ * @author Martin Flower
+ * @author Keisuke Kato
+ */
+class Tomcat9xServer extends BaseTomcat7xPlusImpl {
```
Comment (bmuschko) [inline tomcat9x/src/main/groovy/com/bmuschko/gradle/tomcat/embedded/Tomcat9xServer.groovy]: It looks to me as if `Tomcat8xServer`, `Tomcat85xServer` and `Tomcat9xServer` use pretty much the same code. Let's unify these classes. I believe the only difference is the return value of `getVersion()`. Could you break out another parent class? In the long we'll probably want to rather go for a delegation pattern.

File: tomcat9x/src/integTest/groovy/com/bmuschko/gradle/tomcat/embedded/Tomcat9xServerIntegrationTest.groovy [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+import com.bmuschko.gradle.tomcat.embedded.fixture.EmbeddedTomcatIntegrationTest
+
+/**
+ * Tomcat 9x server test.
+ *
+ * @author Benjamin Muschko
+ * @author Keisuke Kato
+ */
+class Tomcat9xServerIntegrationTest extends EmbeddedTomcatIntegrationTest {
```
Comment (bmuschko) [inline tomcat9x/src/integTest/groovy/com/bmuschko/gradle/tomcat/embedded/Tomcat9xServerIntegrationTest.groovy]: All `EmbeddedTomcatIntegrationTest` long  very similar. We should try to unify the code here as well.

File: tomcat9x/src/main/groovy/com/bmuschko/gradle/tomcat/embedded/Tomcat9xServer.groovy (hunk 1 above)
Comment (k--kato) [reply tomcat9x/src/main/groovy/com/bmuschko/gradle/tomcat/embedded/Tomcat9xServer.groovy]: @bmuschko I created BaseTomcat8xPlusImpl.

File: tomcat9x/src/integTest/groovy/com/bmuschko/gradle/tomcat/embedded/Tomcat9xServerIntegrationTest.groovy (hunk 2 above)
Comment (k--kato) [reply tomcat9x/src/integTest/groovy/com/bmuschko/gradle/tomcat/embedded/Tomcat9xServerIntegrationTest.groovy]: @bmuschko I created `EmbeddedTomcat7xPlusIntegrationTest`.

**Comment by k--kato** [comment]
@bmuschko This is the stack trace on [Travis CI](https://travis-ci.org/bmuschko/gradle-tomcat-plugin/builds/282312903?utm_source=github_status&utm_medium=notification).

```java
java.net.ConnectException: Connection refused (Connection refused)

	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at java.net.Socket.connect(Socket.java:538)
	at java.net.Socket.<init>(Socket.java:434)
	at java.net.Socket.<init>(Socket.java:244)
	at com.bmuschko.gradle.tomcat.embedded.fixture.EmbeddedTomcatIntegrationTest.verifyCreatedSocket(EmbeddedTomcatIntegrationTest.groovy:30)
	at com.bmuschko.gradle.tomcat.embedded.fixture.EmbeddedTomcatIntegrationTest.Can start server(EmbeddedTomcatIntegrationTest.groovy:56)
```

**Comment by k--kato** [comment]
@bmuschko I self-solved the test error.

**Comment by k--kato** [comment]
@bmuschko Could you review this? I'd like to use this great plugin on Java 9.

**Comment by bmuschko** [comment]
@k--kato Thanks for touching up the changes. I am going to review them soon.

**Comment by kasecato** [comment]
@bmuschko Thanks for the review and merging 🎉

**Comment by bmuschko** [comment]
@kasecato You are welcome!

**Pull request merged**